### PR TITLE
PT test: Add 5 PT cases.

### PR DESCRIPTION
**Pull request by ammyyi**
Signed-off-by: ammy.yi@intel.com <ammy.yi@intel.com>

### CODE REVIEW

File: pt/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+## Build Tips
+Need prepare intel-pt.h and libipt.so for building PT related cases.
+You can get them in https://github.com/intel/libipt.
+
+## Usage
+Need libipt.so.2 and add it into LD_LIBRARY_PATH before running PT cases.
+You can build libipt.so.2 in https://github.com/intel/libipt.
+
+./branch
+Disable branch and check no TIP/FUP/TNT.
```
Comment (hanning0511) [inline pt/README.md:16]: Better to put this in code block, for example

```
# Disable branch and check no TIP/FUP/TNT.
./branch
```

Same for the following parts.

File: pt/branch.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#include "utils.h"
+
+/**
+ * disable branch check :
+ *	disbale branch and check package
+ *	TIP/FUP/TNT: should not be expected
+ *	check basic expected packet inside traces
+ *	PASS will return 0 and FAIL will return 1
+ */
+int disablebranchcheck(void)
```
Comment (ysun) [inline pt/branch.c:13]: This func name can be disable_branch_check according to the naming regulation.

File: pt/utils.c [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	if (type == -1)
+		printf(f ? "PT pmu has a wtf type\n"
+			: "PT pmu not found\n");
+
+	return type;
+}
+/*
+ * initialization event attribut
+ */
+void iniEvtAttr(struct perf_event_attr *attr)
```
Comment (ysun) [inline pt/utils.c:91]: Do not use Camel-Case naming. All the function reference to this function should be updated.

File: pt/branch.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	if (seekPck_w_lib(c_type, bufM[1], head) == 0) {
+		printf("TNT should not be found!\n");
+		FAIL = 1;
+	}
+
+	/* unmap and close */
+	delMap(bufM, bufSz, 1, fdi);
+	close(fde);
+onerror:
+	printf("USER trace address check %s\n", FAIL ? "FAIL" : "PASS");
```
Comment (ysun) [inline pt/branch.c:85]: From my record, we have a rule for the results' info print. e.g.
[PASS]case_name: comments
[FAIL]case_name: comments

File: pt/branch.c [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+ *	check basic expected packet inside traces
+ *	PASS will return 0 and FAIL will return 1
+ */
+int main(int argc, char *argv[])
+{
+	unsigned int CASEID;
+	int result = 0;
+
+	result = disablebranchcheck();
+	printf("CASE result = %d\n", result);
```
Comment (ysun) [inline pt/branch.c:103]: Please use this format:
[PASS]case_name: comments

File: pt/utils.c [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+	attr->size			= sizeof(struct perf_event_attr);
+	attr->exclude_kernel		= 0;
+	attr->exclude_user		= 0;
+	attr->mmap			= 1;
+}
+
+/*
+ * mapping management
+ */
+uint64_t **creaMap(int fde, long bufsize, int SnFuSm, int *fdi)
```
Comment (ysun) [inline pt/utils.c:107]: Please change the function name, like create_map, cr_map, or sth else.

File: pt/utils.c [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+#define RTIT_CTL_TSC_EN         BIT(10)
+#define PT_CTL_DISRETC RTIT_CTL_DISRETC
+#ifndef PT_CTL_TSC_EN
+#define PT_CTL_TSC_EN RTIT_CTL_TSC_EN
+#define PT_CTL_DISRETC RTIT_CTL_DISRETC
+#endif
+
+#define PT_PMU_DIR "/sys/devices/intel_pt/"
+
+int seekPck_w_lib(enum pt_packet_type pt_type, uint64_t *bufEv, long bufsize)
```
Comment (ysun) [inline pt/utils.c:17]: Naming rule...

File: pt/utils.c [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+	if (bufEv[0] != MAP_FAILED && SnFuSm != 2) {
+		pc = bufEv[0];
+		pc->aux_offset = pbufSz;
+		pc->aux_size = bufsize;
+		bufEv[1] = mmap(NULL, bufsize, PROTo, MAP_SHARED, fde, pbufSz);
+	}
+	return bufEv;
+}
+
+void delMap(uint64_t **bufEv, long bufsize, int SnFuSm, int fdi)
```
Comment (ysun) [inline pt/utils.c:140]: Naming rule.

File: pt/cpl.c [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+	return FAIL;
+}
+
+
+/**
+ * kernel mode trace check :
+ *  will set exclude_user=1, so only can get kernel trace logs
+ *	PASS will return 0 and FAIL will return 1
+ */
+int kernelmodecheck(void)
```
Comment (ysun) [inline pt/cpl.c:97]: How about change the name to kernel_mode_check?

File: pt/cpl.c [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+	c_type = ppt_fup;
+	if (seekPck_w_lib(c_type, bufM[1], head) != 0) {
+		printf("check address fup failed\n");
+		FAIL = 1;
+	}
+	/* unmap and close */
+	delMap(bufM, bufSz, 1, fdi);
+	close(fde);
+onerror:
+	printf("KERNEL trace address check  %s\n", FAIL ? "FAIL":"PASS");
```
Comment (ysun) [inline pt/cpl.c:174]: [FAIL]case_name: comments

File: pt/nonroot_test.c [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+	/* stop tracing */
+	if(ioctl(fde, PERF_EVENT_IOC_DISABLE) != 0) {
+		printf("ioctl with PERF_EVENT_IOC_DISABLE is failed!\n");
+		FAIL = 1;
+	}
+
+	printf("bufSz = %ld\n", bufSz);
+	pmp = (struct perf_event_mmap_page * )bufM[0];
+  head = (*(volatile uint64_t *)&(pmp->aux_head));
```
Comment (ysun) [inline pt/nonroot_test.c:70]: You missed a 'tab' here.

File: pt/nonroot_test.c [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+	printf("head = %ld\n", head);
+	if (head == 0){
```
Comment (ysun) [inline pt/nonroot_test.c:72]: Request a 'space' after the '('.

File: pt/nonroot_test.c [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+		FAIL = 1;
+		printf("No trace generated!\n");
+	}
+
+	/* unmap and close */
+	delMap(bufM, bufSz, 1, fdi);
+	close(fde);
+onerror:
+	printf("non priviledge trace check %s\n", FAIL ? "FAIL":"PASS");
```
Comment (ysun) [inline pt/nonroot_test.c:81]: [FAIL]case_name: comments

File: pt/psb.c [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+/**
+ * PSB package check
+ */
+int main(void)
+{
+	int result = 0;
+
+	result = psbcheck();
+	printf("CASE result = %d\n", result);
```
Comment (ysun) [inline pt/psb.c:87]: [PASS/FAIL]case_name: comments

File: pt/utils.c [hunk 15]
```diff
@@ -0,0 +1,165 @@
...
+	pt_config_init(&config);
+	config.begin = bufEv;
+	config.end = bufEv + bufsize;
+
+	decoder = pt_pkt_alloc_decoder(&config);
+	if (!decoder) {
+		printf("pt_pkt_alloc_decoder is failed!\n");
+		r_val = -1;
+		return r_val;
+	}	else {
```
Comment (ysun) [inline pt/utils.c:35]: Replace the tab with a space in the front of 'else', please.
BTW, the else is not necessary if there's a 'return' in the statement of if.

File: pt/utils.c [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+	for (;;) {
+		printf("offset = %d\n", offset);
+		errcode = pt_pkt_get_offset(decoder, &offset);
+		if (errcode < 0) {
+			printf("error getting offset %llu %d\r\n", offset, errcode);
+				r_val = -1;
+				return r_val;
+		}
+		errcode = pt_pkt_next(decoder, &packet, sizeof(packet));
+		if (errcode < 0) {
```
Comment (ysun) [inline pt/utils.c:53]: This condition, I feel, is useless.
Furthermore, this is logically dead code that could never be executed.

File: pt/utils.c [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+	return r_val;
+}
+
+/*
+ * Intel_pt pmu
+ */
+int PT_pmu_type(void)
+{
+	FILE *f = fopen(PT_PMU_DIR "type", "r");
+	int type = -1;
```
Comment (ysun) [inline pt/utils.c:74]: I believe you don't use the var type correctly here. 
It's useless if the 'type' never change but always '-1'.

**Review by ysun — CHANGES_REQUESTED** [review]
And could you please check this patch set with linux/scripts/checkpatch.pl ?

**Pull request merged**